operator/v1/types.go

```diff
@@ -45,6 +45,21 @@ type ClusterManagerSpec struct {
 	// NodePlacement enables explicit control over the scheduling of the deployed pods.
 	// +optional
 	NodePlacement NodePlacement `json:"nodePlacement,omitempty"`
+
+	// HubDeployMode represents the mode used to deploy hub components. There are two mode: mode `Unite` as default, and mode `Separate`.
+	// In Separate mode, a "shell-hub" holds all CRDs, and "worker-hub" has all deployment running in it.
+	// +kubebuilder:default=Unite
+	HubDeployMode string `json:"hubDeployMode"`
+
+	// HubInstalledNamespace represents the desired namespace used to install all hub components.
+	// +optional
+	// +kubebuilder:default=open-cluster-management-hub
+	HubInstalledNamespace string `json:"hubInstalledNamespace,omitempty"`
```

Comment:
put this and kubeconfig in a struct. Do not add hub in the field name.

I would think you have things like

```go

type DeployOption {
    DeployMode string // it can be Default or ExternalController. External controller indicates the controller is deployed in another cluster but not the hub cluster.

    ExternalController ExternalControllerConfig
}

type ExternalControllerConfig struct {
    InstallNamespace
}
```

Do we need to specify hubkubeconfig in API, I think it can be a fixe name with fixed key in it.

Reply:
Sure, it's a better way with a fixed name and fixed key.